File: hw/vizcaino/u3/hw23Singleton/hw23SingletonJavaScript/usTax.js
class USTax {
    constructor() {
        if (USTax.instance) {
            return USTax.instance;
        }
        this.taxRate = 0.0825;
        USTax.instance = this;
    }

    static getInstance() {
        return new USTax();
    }

    getTaxRate() {
        return this.taxRate;
    }

    setTaxRate(taxRate) {
        this.taxRate = taxRate;
    }

    calculateTax(amount) {
        return amount * this.taxRate;
    }
}

module.exports = USTax;

File: ws/vizcaino/u3/ws29Singleton/wsSingleton/src/main/java/ec/edu/espe/wssingleton/UsTax.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.wssingleton;

/**
 *
 * @author Adrian Vizcaino <The-Softwarriors at ESPE>
 */
public class UsTax {

    private static UsTax instance;
    private double taxRate;

    private UsTax() {
        taxRate = 0.0825;
    }

    public static synchronized UsTax getInstance() {
        if (instance == null) {
            instance = new UsTax();
        }
        return instance;
    }

    public double getTaxRate() {
        return taxRate;
    }

    public void setTaxRate(double taxRate) {
        this.taxRate = taxRate;
    }

    public double calculateTax(double amount) {
        return amount * taxRate;
    }
}